app/assets/javascripts/components/story/StoryDescription.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import Parser from 'html-react-parser'
+
+import StoryLink from 'components/story/StoryLink';
+
+const LOCAL_STORY_REGEXP = /(?!\s|\b)(#\d+)(?!\w)/g;
+
+class StoryDescription extends React.PureComponent {
+
+  constructor(props) {
+    super(props);
+
+    const description = Parser(props.description, { replace: domNode => {
+      if (!domNode.attribs || !domNode.attribs['data-story-id']) return;
+
+      const id = domNode.attribs['data-story-id'];
+      const story = props.linkedStories[id];
+      return ( <StoryLink story={story} key={id} /> );
+    }});
+
+    this.state = {
+      description: description,
+      isEmpty: (!props.description || props.description.length === 0)
+    };
+
+    this.renderDescription = this.renderDescription.bind(this);
+    this.parseStoryLinks = this.renderDescription.bind(this);
+  }
+
+  shouldComponentUpdate() {
```

Comment:
Will this never update? If so, why use `React.Component` that already has an implementation of `shouldComponentUpdate` different than that? https://facebook.github.io/react/docs/react-api.html#react.purecomponent. You can just have a functional component and use memoization instead.